libs/audit-service/src/main/java/uk/gov/di/ipv/core/library/auditing/helpers/AuditExtensionsHelper.java

```diff
@@ -1,130 +1,122 @@
+                return new AuditRestrictedF2F(name, docExpiryDate);
+            }
+
+            var drivingPermit = credentialSubject.getDrivingPermit();
+            if (!isNullOrEmpty(drivingPermit)) {
+                var docExpiryDate = drivingPermit.get(0).getExpiryDate();
+                return new AuditRestrictedF2F(name, docExpiryDate);
+            }
+
+            var brp = credentialSubject.getResidencePermit();
+            if (!isNullOrEmpty(brp)) {
+                var docExpiryDate = brp.get(0).getExpiryDate();
+                return new AuditRestrictedF2F(name, docExpiryDate);
+            }
+
+            var idCard = credentialSubject.getIdCard();
+            if (!isNullOrEmpty(idCard)) {
+                var docExpiryDate = idCard.get(0).getExpiryDate();
+                return new AuditRestrictedF2F(name, docExpiryDate);
+            }
+
+            return new AuditRestrictedF2F(name);
+        } else {
+            LOGGER.warn(LogHelper.buildLogMessage("VC not of type IdentityCheckCredential."));
+            return new AuditRestrictedF2F(null);
+        }
     }
 
     public static AuditRestrictedInheritedIdentity getRestrictedAuditDataForInheritedIdentity(
-            VerifiableCredential vc, String deviceInformation) throws CredentialParseException {
-        var credentialSubject =
-                OBJECT_MAPPER
-                        .valueToTree(vc.getClaimsSet())
-                        .path(CLAIMS_CLAIM)
-                        .path(VC_CLAIM)
-                        .path(VC_CREDENTIAL_SUBJECT);
-
-        try {
+            VerifiableCredential vc, String deviceInformation)
+            throws HttpResponseExceptionWithErrorBody {
+        if (vc.getCredential() instanceof IdentityCheckCredential identityCheckCredential) {
+            var credentialSubject = getCredentialSubjectOrThrow(identityCheckCredential);
+
             return new AuditRestrictedInheritedIdentity(
-                    OBJECT_MAPPER.treeToValue(credentialSubject.path(VC_NAME), LIST_NAME_TYPE),
-                    OBJECT_MAPPER.treeToValue(
-                            credentialSubject.path(VC_BIRTH_DATE), LIST_BIRTH_DATE_TYPE),
-                    OBJECT_MAPPER.treeToValue(
-                            credentialSubject.path(VC_SOCIAL_SECURITY_RECORD),
-                            LIST_SOCIAL_SECURITY_RECORD_TYPE),
+                    credentialSubject.getName(),
+                    credentialSubject.getBirthDate(),
+                    credentialSubject.getSocialSecurityRecord(),
                     new DeviceInformation(deviceInformation));
-        } catch (JsonProcessingException e) {
-            throw new CredentialParseException("Unable to parse VC for inherited ID audit data", e);
+        } else {
+            LOGGER.error(LogHelper.buildLogMessage("VC must be of type IdentityCheckCredential."));
+            throw new HttpResponseExceptionWithErrorBody(
```

Comment:
Similarly, can this be a warning only - we don't really want to fail on an audit log.